pipert/core/message_handlers.py

```diff
@@ -48,13 +74,47 @@ def __init__(self, url, maxlen=100):
         self.conn = None
         self.url = url
         self.maxlen = maxlen
+        self.last_msg_id = None
         self.connect()
 
+    def read_next_msg(self, in_key):
+        if self.last_msg_id:
+            last_msg_id_to_read = \
+                self._add_offset_to_stream_id(self.last_msg_id, 1)
+        else:
+            return self.receive(in_key)
+
+        redis_msg = self.conn.xrange(in_key,
+                                     count=1,
+                                     min=last_msg_id_to_read)
+        if not redis_msg:
+            return None
+        self.last_msg_id = redis_msg[0][0].decode()
+        msg = redis_msg[0][1]["msg".encode("utf-8")]
+        return msg
+
+    def read_most_recent_msg(self, in_key):
+        if self.last_msg_id:
+            last_msg_id_to_read = \
+                self._add_offset_to_stream_id(self.last_msg_id, 1)
+        else:
+            return self.receive(in_key)
+
+        redis_msg = self.conn.xrevrange(in_key,
+                                        count=1,
+                                        min=last_msg_id_to_read)
+        if not redis_msg:
+            return None
+        self.last_msg_id = redis_msg[0][0].decode()
+        msg = redis_msg[0][1]["msg".encode("utf-8")]
+        return msg
+
```

Comment:
The functions are identical except for `redis_msg = ...`.
You can make a third private function that would get the redist msg according to a parameter sent to it instead of this code duplication.